spec/models/user_spec.rb

```diff
@@ -1,5 +1,73 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  # pending "add some examples to (or delete) #{__FILE__}"
+
+  let(:user) { 
+               User.create!(full_name: "R.L. Stine",
+                            display_name: "Stiner",
+                            email: "goosebumps@rscary.com",
+                            password: "aaah")
+  }
+  
+  it "is valid" do
+    expect(user).to be_valid
+  end
+  
+  it "has valid attributes" do
+    expect(user.full_name).to eq("R.L. Stine")
+    expect(user.display_name).to eq("Stiner")
+    expect(user.email).to eq("goosebumps@rscary.com")
+    expect(user.password).to eq("aaah")
+  end
+  
+  it "must have valid email" do
+    user.email = "thisemailsucks"
+    expect(user).to_not be_valid
+
+    user.email = "thisemailsucks.com"
+    expect(user).to_not be_valid
+
+    user.email = "thisemailsucks@nocom"
+    expect(user).to_not be_valid
+    
+    user.email = "this email sucks@email.com"
+    expect(user).to_not be_valid
+
+    user.email = ""
+    expect(user).to_not be_valid
+  end
+  
+  it "must have unique email" do
+    user
+    expect { User.create!(full_name: "R.L. NotStine",
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.